oemer/staffline_extraction.py

```diff
@@ -331,51 +331,53 @@ def extract(
 
     # Start process
     zones, *_ = init_zones(staff_pred, splits=splits)
-    all_staffs = []
+    all_staffs: List[List[Staff]] = []
     for rr in zones:
         print(rr[0], rr[-1], end=' ')
         rr = np.array(rr, dtype=np.int64)
         staffs = extract_part(staff_pred[:, rr], x_offset=rr[0], line_threshold=line_threshold)
         if staffs is not None:
             all_staffs.append(staffs)
             print(len(staffs))
-    all_staffs = align_staffs(all_staffs)  # type: ignore
+    aligned_staffs: np.ndarray = align_staffs(all_staffs)
 
     # Use barline information to infer the number of tracks for each group.
-    num_track = further_infer_track_nums(all_staffs, min_degree=barline_min_degree)  # type: ignore
+    num_track = further_infer_track_nums(aligned_staffs, min_degree=barline_min_degree)
     logger.debug(f"Tracks: {num_track}")
-    for col_sts in all_staffs:
+    for col_sts in aligned_staffs:
         for idx, st in enumerate(col_sts):
             st.track = idx % num_track
             st.group = idx // num_track
 
     # Validate staffs across zones.
     # Should have same number of staffs
-    if not all([len(staff) == len(all_staffs[0]) for staff in all_staffs]):
+    if not all([len(staff) == len(aligned_staffs[0]) for staff in aligned_staffs]):
         raise Exception
-    assert all([len(staff) == len(all_staffs[0]) for staff in all_staffs])
+    assert all([len(staff) == len(aligned_staffs[0]) for staff in aligned_staffs])
 
     norm = lambda data: np.abs(np.array(data) / np.mean(data) - 1)
-    for staffs in all_staffs.T:  # type: ignore
+    valid_staffs: list[list[Staff]] = []
+    for staffs in aligned_staffs.T:
         # Should all have 5 lines
         line_num = [len(staff.lines) for staff in staffs]
         if len(set(line_num)) != 1:
-            raise E.StafflineCountInconsistent(
-                f"Some of the stafflines contains less or more than 5 lines: {line_num}")
+            print(f"Some of the stafflines contains less or more than 5 lines: {line_num}")
+            continue
```

Comment:
I'm not sure if ignoring these exceptions is good or not, since the later flow relies heavily on strong stafflines assumptions.

Reply:
In the cases I saw, with the change you get e.g. 4 out of 5 staffs. The results then weren't great. I think what the change mainly allows is to understand better which staff (or part of the image) caused the problems. Like you can see in the teaser, what staffs have been detected and which ones haven't. With the exception, it's much harder to understand that as you also don't get a teaser image.

But that said, I'm also fine to revert this change.

Reply:
Maybe you can do some tests on those cases with and without these modifications to see if removing the exception really effects the later process?